lib/oidcstrategy.js

```diff
@@ -752,4 +577,494 @@ Strategy.prototype._shouldLoadUserProfile = function shouldLoadUserProfile(issue
+                  refresh_token,
+                  params,
+                });
+              });
+            }
+
+            // lets do an id_token fallback. We use id_token over userInfo endpoint for now
+            return onProfileLoaded(self, {
+              req,
+              sub,
+              iss,
+              profile: makeProfileObject(jwtClaims, jwtClaimsStr),
+              jwtClaims,
+              access_token,
+              refresh_token,
+              params,
+            });
+          });
+        });
+      });
+  });
+};
+
+/**
+ * prepare the initial authorization request
+ *
+ * @params {Object} req
+ * @params {Function} next
+ */
+Strategy.prototype._flowInitializationHandler = function flowInitializationHandler(req, next) {
+  // The request being authenticated is initiating OpenID Connect
+  // authentication. Prior to redirecting to the provider, configuration will
+  // be loaded. The configuration is typically either pre-configured or
+  // discovered dynamically. When using dynamic discovery, a user supplies
+  // their identifer as input.
+
+  const self = this;
+
+  log.info(`we are in the OpenID Connect Inital Flow`);
+
+  let identifier;
+  if (req.body && req.body[this._identifierField]) {
+    identifier = req.body[this._identifierField];
+  } else if (req.query && req.query[this._identifierField]) {
+    identifier = req.query[this._identifierField];
+  }
+
+  return self.configure(identifier, (configureError, config) => {
+    if (configureError) {
+      return next(configureError);
+    }
+
+    var options = self._options;
+
+    let callbackURL = options.callbackURL || config.callbackURL;
+    if (callbackURL) {
+      const parsed = url.parse(callbackURL);
+      if (!parsed.protocol) {
+        // The callback URL is relative, resolve a fully qualified URL from the
+        // URL of the originating request.
+        callbackURL = url.resolve(utils.originalURL(req), callbackURL);
+      }
+    }
+
+    log.info('Going in with our config loaded as: ', config);
+
+    const params = {};
+    if (self.authorizationParams) {
+      _.assign(params, self.authorizationParams(options));
+    }
+    _.assign(params, {
+      redirect_uri: callbackURL,
+    }, objectTransform({
+      source: config,
+      map: {
+        responseMode: 'response_mode',
+        responseType: 'response_type',
+        clientID: 'client_id',
+        resourceURL: 'resource',
+      },
+    }));
+
+    log.info('We are sending the response_type: ', params.response_type);
+    log.info('We are sending the response_mode: ', params.response_mode);
+
+    let scope = config.scope;
+    if (Array.isArray(scope)) {
+      scope = scope.join(config.scopeSeparator);
+    }
+    if (scope) {
+      params.scope = ['openid', scope].join(config.scopeSeparator);
+    } else {
+      params.scope = 'openid';
+    }
+
+    // if (policy) { params['p'] = policy; }; // Policy parameter should be included.
```

Comment:
Not sure what your team's practices are, but should this have a `TODO:` or `MUSTFIX:` prefix in a comment, or anything, or is this commented out long-term by design?